accept.go

```diff
@@ -349,12 +349,18 @@ func websocketExtensions(h http.Header) []websocketExtension {
 }
 
 func headerTokens(h http.Header, key string) []string {
+	return headerTokensExt(h, key, true)
+}
+
+func headerTokensExt(h http.Header, key string, lower bool) []string {
 	key = textproto.CanonicalMIMEHeaderKey(key)
 	var tokens []string
 	for _, v := range h[key] {
 		v = strings.TrimSpace(v)
 		for _, t := range strings.Split(v, ",") {
-			t = strings.ToLower(t)
+			if lower {
```

Comment:
Perhaps we shouldn't do a `ToLower` here at all and instead rely on the caller handling case correctly?

Reply:
I see this code is also going to cause trouble for `Sec-WebSocket-Extensions`. In general, I don't think HTTP header values are case insensitive so I think I made a mistake here.

Reply:
I think `headerContainsToken` should be the one modified to allow optional case insensitivity as that's the function used to check `Connect` and `Upgrade` whose values are case insensitive. And then this function should be changed to not modify case at all.

Reply:
Rewrote the fix. Decided to rename `headerContainsToken` to `headerContainsTokenIgnoreCase` because this function is only used to deal with `Connect` and `Upgrade` headers.